src/backends/plonky2/primitives/merkletree.rs

```diff
@@ -260,6 +264,99 @@ impl MerkleProof {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub struct MerkleClaimAndProof {
+    /// `enabled` determines if the merkleproof verification is enabled
+    pub enabled: bool,
```

Comment:
Initially I was not sure on having this boolean in the proof structure, but I understand that is in order to do stuff like: https://github.com/0xPARC/pod2/tree/cedd98dbe10433b0c030f1b0916485cef3773a90/src/backends/plonky2/circuits/mainpod.rs#L144 .
The concern was about devs missusing the merkletree module (like verifying a disabled proof as it were enabled), but I understand that this is not the target of the module, as the target usage of the merkletree module is not for external usage but only for internal usage by other modules of POD, right?

Reply:
> The concern was about devs missusing the merkletree module (like verifying a disabled proof as it were enabled), but I understand that this is not the target of the module, as the target usage of the merkletree module is not for external usage but only for internal usage by other modules of POD, right?

That's right, the target usage is internal.  Are you thinking about removing the `enabled` from `MerkleClaimAndProof` and then having this API for `set_targets`?
```rust
impl MerkleClaimAndProofTarget {
    /// assigns the given values to the targets
    pub fn set_targets(
        &self,
        pw: &mut PartialWitness<F>,
        enabled: bool,
        mp: &MerkleClaimAndProof,
    ) -> Result<()> {
```

Perhaps that's more clear?

Reply:
Doing the logic at https://github.com/0xPARC/pod2/tree/cedd98dbe10433b0c030f1b0916485cef3773a90/src/backends/plonky2/circuits/mainpod.rs#L144 with the `enabled` outside would be more cumbersome because currently we're including the `enabled` in the random access.
But we could just force that all merkle proofs have enabled=true and fill them with valid proofs and then we don't need to add `enabled` into the random access.

Reply:
Thanks for the explanation ^^
I don't have any strong opinion, and agree with that removing the `enabled` field from the claimandproof struct would add a bit of extra overhead in the code. So, happy with the way it is.